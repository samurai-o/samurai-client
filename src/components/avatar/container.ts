import styled from 'styled-components';

export const AvatarContainer = styled.div`
  border-radius: 10px;
  cursor: pointer;
  overflow: hidden;
`;
