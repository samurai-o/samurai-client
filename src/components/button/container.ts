import styled from 'styled-components';
import { ButtonType, colors } from '.';

export type ButtonStyledContainer = {
  color: ButtonType;
};

export const ButtonContainer = styled.button<ButtonStyledContainer>`
  padding: 4px 10px;
  border-radius: 4px;
  border: transparent;
  outline: none;
  cursor: pointer;
  color: ${({ color }) => (color !== 'text' ? '#fff' : '#000')};
  background: ${({ color }) => colors[color]};
`;
