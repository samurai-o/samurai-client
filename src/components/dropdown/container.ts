import styled from 'styled-components';
import { Placements } from '.';

export const DropdownContainer = styled.div`
  position: relative;
`;

export type DropdownBoxStyleProps = {
  show?: boolean;
  placements: Placements;
};
export const DropdownBoxContainer = styled.div<DropdownBoxStyleProps>`
  min-width: 100px;
  position: absolute;
  top: calc(100% + 2px);
  padding-top: 4px;
  padding-bottom: 4px;
  background: #fff;
  left: ${({ placements }) =>
    placements === 'left'
      ? '0px'
      : placements === 'center'
      ? 'calc(50% - 4px)'
      : 'unset'};
  right: ${({ placements }) => (placements === 'right' ? '0px' : 'unset')};
  transition: opacity 250ms ease, transform 250ms ease;
  box-shadow: 0px 4px 8px rgb(0 0 0 / 10%);
  backface-visibility: hidden;
  border-radius: 4px;
  border: 1px solid #d0d7de;
  pointer-events: ${({ show }) => (!!show ? 'all' : 'none')};
  opacity: ${({ show }) => Number(!!show)};
  transform: ${({ show }) =>
    !!show
      ? 'translate3d(0, 0, 0)'
      : 'translate3d(0, 32px, 0) rotateX(-16deg) rotateY(14deg)'};
  &::after {
    position: absolute;
    bottom: 100%;
    left: ${({ placements }) =>
      placements === 'left'
        ? '8px'
        : placements === 'center'
        ? 'calc(50% - 4px)'
        : 'unset'};
    right: ${({ placements }) => (placements === 'right' ? '8px' : 'unset')};
    content: '';
    transform: translateY(4px) rotate(45deg);
    width: 8px;
    height: 8px;
    border: 1px solid #d0d7de;
    border-right: none;
    border-bottom: none;
    background-color: #fff;
  }
`;
