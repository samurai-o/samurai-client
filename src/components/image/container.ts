import styled from 'styled-components';

export const ImageContainer = styled.div<any>`
  width: ${({ width }) => (typeof width === 'number' ? `${width}px` : '100px')};
  border-radius: 8px;
  overflow: hidden;
`;

export const ImageDom = styled.img`
  width: inherit;
  vertical-align: middle;
`;
