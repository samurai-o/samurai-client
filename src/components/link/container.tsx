import styled from 'styled-components';

export const LinkButton = styled.a`
  width: fit-content;
  text-decoration: none;
  display: block;
  cursor: pointer;
`;
