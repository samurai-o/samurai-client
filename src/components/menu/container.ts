import styled from 'styled-components';
import { Overall } from '.';

export const MenuContainer = styled.div``;

export type MenuListStyleProps = {
  noEvent?: boolean;
  overall: Overall;
};

export const MenuList = styled.ul<MenuListStyleProps>`
  list-style: none;
  margin: unset;
  padding: unset;
  display: flex;
  flex-direction: ${({ overall }) =>
    overall === 'vertical' ? 'column' : 'row'};
`;

export type MenuItemStyleProps = {
  noEvent?: boolean;
  tag?: string | number;
};

export const MenuItem = styled.li<MenuItemStyleProps>`
  list-style: none;
  margin: unset;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 4px;
  overflow: hidden;
  margin: 0 4px;
  transition: all 250ms ease;
  &:hover {
    background: #e0dede47;
  }
  &:active {
    background: #ddd;
  }
`;
