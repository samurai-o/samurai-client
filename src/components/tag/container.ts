import styled from 'styled-components';

const tagcontainerbackground: { [key in TagContainerBackground]: string } = {
  default: '#fff',
  red: '#e21253e0s',
  blue: 'blue',
};

export type TagContainerBackground = 'red' | 'blue' | 'default';

export type TagContainerProps = {
  background?: TagContainerBackground;
};
export const TagContainer = styled.span<TagContainerProps>`
  cursor: pointer;
  display: inline-flex;
  padding: 0px 6px;
  border-radius: 4px;
  border: 1px solid #ddd;
  background: ${(props) =>
    !props.background
      ? tagcontainerbackground['default']
      : tagcontainerbackground[props.background]};
`;
