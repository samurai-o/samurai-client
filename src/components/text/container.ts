import styled from 'styled-components';

export type TextStyleContainer = {
  size?: number;
  blod?: boolean;
  color?: string;
};

export const TitleContainer = styled.h6`
  margin: unset;
`;

export const SubtitleContainer = styled.p<TextStyleContainer>`
  font-size: ${({ size }) => (typeof size === 'number' ? `${size}px` : '12px')};
  font-weight: ${({ blod }) => (!!blod ? '500' : 'unset')};
  color: ${(props) => (!!props.color ? props.color : 'inherit')};
  font-weight: 500;
  margin: unset;
  padding: unset;
`;

export const TextContainer = styled.span<TextStyleContainer>`
  font-size: ${({ size }) => `${size}px`};
  font-weight: ${({ blod }) => (!!blod ? '500' : 'unset')};
  color: ${(props) => (!!props.color ? props.color : 'inherit')};
  margin: unset;
  padding: unset;
`;
