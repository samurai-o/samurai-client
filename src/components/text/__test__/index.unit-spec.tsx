import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Text } from '..';
import { TitleContainer } from '../container';

describe('avatar 组件', () => {
  it('快照测试', () => {
    const node = create(<Text type="title">测试</Text>);
    expect(node).toMatchSnapshot();
  });

  it('子元素渲染验证', () => {
    const node = shallow(<Text />);
    expect(node.isEmptyRender()).toEqual(true);
  });

  it('type 参数切换验证', () => {
    const node = shallow(<Text type="title">测试</Text>);
    expect(node.find(TitleContainer).prop('children')).toEqual('测试');
  });

  it('type 非正确参数', () => {
    const node = shallow(<Text type="body">测试</Text>);
    expect(node.getElement().props).toEqual({ children: null });
  });
});
