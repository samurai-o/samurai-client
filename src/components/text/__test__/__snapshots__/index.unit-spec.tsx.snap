// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar 组件 快照测试 1`] = `
<h6
  className="sc-bdvvaa kuTvfO"
>
  测试
</h6>
`;
